release-calendar/src/client/models/release-event.ts

```diff
@@ -13,69 +13,46 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 import {EventInput} from '@fullcalendar/core/structs/event';
 import {EventSourceInput} from '@fullcalendar/core/structs/event-source';
-import {Release} from '../../types';
-
-function convertReleaseToEvent(release: Release): EventInput {
-  return {
-    title: release.name,
-    start: release.date,
-    className: release.channel,
-    extendedProps: {isRollback: release.isRollback},
-  };
-}
+import {EventTuple, Release} from '../../types';
 
-function groupIntoChannels(events: EventInput[]): EventSourceInput[] {
-  const ltsEvents: EventInput[] = [],
-    stableEvents: EventInput[] = [],
-    percBetaEvents: EventInput[] = [],
-    percExperimentalEvents: EventInput[] = [],
-    optInBetaEvents: EventInput[] = [],
-    optInExperimentalEvents: EventInput[] = [],
-    nightlyEvents: EventInput[] = [];
-
-  events.forEach((event: EventInput) => {
-    switch (event.className) {
-      case 'lts':
-        ltsEvents.push(event);
-        break;
-      case 'stable':
-        stableEvents.push(event);
-        break;
-      case 'perc-beta':
-        percBetaEvents.push(event);
-        break;
-      case 'perc-experimental':
-        percExperimentalEvents.push(event);
-        break;
-      case 'opt-in-beta':
-        optInBetaEvents.push(event);
-        break;
-      case 'opt-in-experimental':
-        optInExperimentalEvents.push(event);
-        break;
-      case 'nightly':
-        nightlyEvents.push(event);
-        break;
-      default:
-        console.log('not logged');
-        break;
-    }
-  });
+function convertReleaseToEvent(release: Release): EventTuple {
   return [
-    {events: ltsEvents, color: 'black', textColor: 'white'},
-    {events: stableEvents, color: 'gray', textColor: 'white'},
-    {events: percBetaEvents, color: 'blue', textColor: 'white'},
-    {events: percExperimentalEvents, color: 'green', textColor: 'white'},
-    {events: optInBetaEvents, color: 'purple', textColor: 'white'},
-    {events: optInExperimentalEvents, color: 'silver', textColor: 'white'},
-    {events: nightlyEvents, color: 'red', textColor: 'white'},
+    release.channel,
+    {
+      title: release.name,
+      start: release.date,
+      className: release.channel,
+      extendedProps: {isRollback: release.isRollback},
+    },
   ];
 }
 
 export function getEvents(releases: Release[]): EventSourceInput[] {
-  return groupIntoChannels(
-    releases.map(release => convertReleaseToEvent(release)),
+  const dict: {[channel: string]: EventInput[]} = {};
+  dict['lts'] = [];
+  dict['stable'] = [];
+  dict['perc-beta'] = [];
+  dict['perc-experimental'] = [];
+  dict['opt-in-beta'] = [];
+  dict['opt-in-experimental'] = [];
+  dict['nightly'] = [];
+
+  const channelEventPairs: EventTuple[] = releases.map(release =>
+    convertReleaseToEvent(release),
   );
+
+  channelEventPairs.forEach(pair => dict[pair[0]].push(pair[1]));
```

Comment:
How about using a map like this - 
```suggestion
const map = new Map();
releases.forEach(release => {
  const event = convertReleaseToEvent(release);
  const channelEvents = map.get(event.className);
  if (!channelEvents) {
    map.set(event.className, [event]);
  } else {
    map.set(event.className, channelEvents.push(event));
  }
});
```

Reply:
Just recommitted with the change to map. Basically used what you suggested with a small adjustment in the else statement. Because .push returns the new length of the array, when trying to add a second event to a channel a 2 would be set instead of the updated array.

Reply:
Oops! If you want, you could also use spread syntax, i.e.
```
const a = [1, 2];
const b = 3;
const c = [...a, b];
console.log(c); // [1, 2, 3]
```

Reply:
oh, Ive never seen that before, but cool!
...is that what you want me to do? or is what i did okay?

what i did being:
channelEvents.push(event);
map.set(event.className, channelEvents);

Reply:
done! I've never seen that syntax before, but it's pretty :)

Reply:
Yeah right? :) It's new as of ES6